File: 2021/08/14/power_of_two.py
"""
Problem: https://leetcode.com/problems/power-of-two/
Author: Andrew Jarombek
Date: 8/14/2021
"""


def is_power_of_two(n: int) -> bool:
    if n == 0:
        return False

    while n != 1:
        if n % 2 == 0:
            n = n // 2
        else:
            return False

    return True


if __name__ == '__main__':
    assert is_power_of_two(16)
    assert not is_power_of_two(17)

File: 2022/07/07/java/PowerOfTwo.java
/**
 * https://leetcode.com/problems/power-of-two/
 * @author Andrew Jarombek
 * @since 7/7/2022
 */

public class PowerOfTwo {
    public static boolean isPowerOfTwo(int n) {
        if (n <= 0)
            return false;

        while (n > 1) {
            if (n % 2 != 0) {
                return false;
            }

            n = n >> 1;
        }

        return true;
    }

    public static void main(String[] args) {
        assert !isPowerOfTwo(0);
        assert isPowerOfTwo(1);
        assert isPowerOfTwo(2);
        assert !isPowerOfTwo(3);
    }
}
